dm/dm/ctl/master/task_operate.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package master
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"sort"
+	"sync"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiflow/dm/dm/ctl/common"
+	"github.com/pingcap/tiflow/dm/openapi"
+	"github.com/spf13/cobra"
+)
+
+const (
+	operateBatchSizeFlag    = "batch-size"
+	defaultOperateBatchSize = 5
+)
+
+type batchTaskOperateResult struct {
+	Result bool                 `json:"result"`
+	Msg    string               `json:"msg"`
+	Tasks  []*taskOperateResult `json:"tasks"`
+}
+
+type taskOperateResult struct {
+	Task   string      `json:"task"`
+	Op     string      `json:"op"`
+	Result bool        `json:"result"`
+	Msg    interface{} `json:"msg"`
+}
+
+func taskOperateFunc(cmd *cobra.Command, operateFunc func(cmd *cobra.Command, taskName string, sources []string) *taskOperateResult) error {
+	argLen := len(cmd.Flags().Args())
+	if argLen == 0 {
+		// may want to operate tasks bound to a source
+		return sourceTaskOperateFunc(cmd, operateFunc)
+	} else if argLen > 1 {
+		// can pass at most one task-name/task-conf
+		cmd.SetOut(os.Stdout)
+		common.PrintCmdUsage(cmd)
+		return errors.New("please check output to see error")
+	}
+
+	taskName := common.GetTaskNameFromArgOrFile(cmd.Flags().Arg(0))
+	sources, err := common.GetSourceArgs(cmd)
+	if err != nil {
+		return err
+	}
+
+	// check task is exist
+	notExist, err := isTaskNotExist(taskName)
```

Comment:
i'm afraid not, if we stop task, there is no check and will return success, It may be considered that there are no subtask need to handle. [here](https://github.com/pingcap/tiflow/blob/0294e283f28bdccce80d3ed2c71b3e735e0923fb/dm/dm/master/scheduler/scheduler.go#L1681). I found it out when I was writing the test too.

Reply:
if so, i think better to change [here](https://github.com/pingcap/tiflow/blob/0294e283f28bdccce80d3ed2c71b3e735e0923fb/dm/dm/master/scheduler/scheduler.go#L1681)  because the coment said

> no subtask need to update, this should not happen

But this actually happened 🤣, my point is if server found user want to operate an object whis is not exist, server should retun an error instead of asking the client to do this verification